addons/binding/org.openhab.binding.wso2iots/src/main/java/org/openhab/binding/wso2iots/handler/wso2iotsHandler.java

```diff
@@ -0,0 +1,390 @@
+            }
+            logger.debug("Update channel {} with state {} ({})", channelId, (state == null) ? "null" : state.toString(),
+                    (iotsResponse == null) ? "null" : iotsResponse.getClass().getSimpleName());
+
+            // Update the channel
+            if (state != null) {
+                updateState(channelId, state);
+            }
+        }
+    }
+
+    private BigDecimal updateBuildingMonitorData(String sensorType) throws Exception {
+
+        BigDecimal iotsResponse = null;
+        boolean resultok = false;
+        String errormsg = null;
+        String response1 = null;
+        wso2iotsConfiguration config = getConfigAs(wso2iotsConfiguration.class);
+        String Access_Token = config.apikey;
+        String id = config.deviceId;
+        // String Access_Token = "bfe52f8a-5f34-382b-802c-93e52468564f";
+        // String id = "2940205";
+        long time = System.currentTimeMillis() / 1000;
+        String fromTime = String.valueOf(time - 200);
+        String toTime = String.valueOf(time);
+
+        String url = "https://localhost:8243/senseme/device/1.0.0/stats/";
+        url = url + id + "?from=" + fromTime + "&to=" + toTime + "&sensorType=" + sensorType;
+
+        URL obj;
+        // if(Boolean.parseBoolean(System.getProperty(""))) {
+        obj = this.validateCertificate(url);
+
+        HttpURLConnection con = (HttpURLConnection) obj.openConnection();
+
+        // optional default is GET
+        con.setRequestMethod("GET");
+
+        // add request header
+        con.setRequestProperty("Accept", "application/json");
+        con.setRequestProperty("Authorization", "Bearer " + Access_Token);
+
+        try {
+
+            int responseCode = con.getResponseCode();
+
+            // logger.debug("\nSending 'GET' request to URL : " + url);
+            // logger.debug("Response Code : " + responseCode);
+
+            BufferedReader in = new BufferedReader(new InputStreamReader(con.getInputStream()));
+            String inputLine;
+            StringBuffer response = new StringBuffer();
+
+            while ((inputLine = in.readLine()) != null) {
+                response.append(inputLine);
+            }
+            in.close();
+
+            // print result
+            // System.out.println(response.toString());
+
+            response1 = response.toString();
+
+            if (response1.equals("[]")) {
+                resultok = false;
+                errormsg = "empty response from wso2 iots";
+                logger.warn(errormsg);
+                // return null;
+
+            } else {
+                String segment = response1.substring(response1.lastIndexOf("{"), response1.lastIndexOf("}"));
+                segment = "{\"values\":" + segment + "}";
+
+                logger.debug(response1);
+                logger.debug(segment);
+                // Map the JSON response to an object
+
+                if (sensorType == wso2iotsBindingConstants.TEMPERATURE) {
```

Comment:
Is it possible to query all sensors altogether?

Reply:
building monitor plugin has the API to get sensor data separately